packages/drivers/routerlicious-driver/src/restWrapper.ts

```diff
@@ -44,6 +40,49 @@ const axiosRequestConfigToFetchRequestConfig = (
 	return [requestInfo, requestInit];
 };
 
+export interface IR11sResponse<T> {
+	content: T;
+	headers: Map<string, string>;
+	propsToLog: ITelemetryProperties;
+	requestUrl: string;
+}
+
+function headersToMap(headers: Headers) {
+	const newHeaders = new Map<string, string>();
+	for (const [key, value] of headers.entries()) {
+		newHeaders.set(key, value);
+	}
+	return newHeaders;
+}
+
+export function getPropsToLogFromResponse(headers: {
+	get: (id: string) => string | undefined | null;
+}) {
+	interface LoggingHeader {
+		headerName: string;
+		logName: string;
+	}
+
+	// We rename headers so that otel doesn't scrub them away. Otel doesn't allow
+	// certain characters in headers including '-'
+	const headersToLog: LoggingHeader[] = [
+		{ headerName: "x-correlation-id", logName: "clientCorrelationId" },
+		{ headerName: "content-encoding", logName: "contentEncoding" },
+		{ headerName: "content-type", logName: "contentType" },
+	];
+	const additionalProps: ITelemetryProperties = {
+		contentsize: TelemetryLogger.numberFromString(headers.get("content-length")),
```

Comment:
Why it's special cased (and not in above array)?

Reply:
Because we wanted to convert it to a number unlike others. This is how we do it in odsp also. This keeps below if to be simple.
TelemetryLogger.numberFromString(headers.get("content-length"))